### ISSUE TITLE
Consider removing useless `-sign-ext` flag in substrate-wasm-builder 

### ISSUE THREAD

**Issue by StackOverflowExcept1on** [paritytech/polkadot-sdk#5777]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

https://github.com/rust-lang/blog.rust-lang.org/blob/4a72d620768932213baf39b03c6f1e61c00cd038/posts/2024-08-26-webassembly-targets-change-in-default-target-features.md

### Request

Should be removed here
https://github.com/paritytech/polkadot-sdk/blob/c8d5e5a383c01ee02c3cc49fbd5e07540b6b79cc/substrate/utils/wasm-builder/src/wasm_project.rs#L836-L842

### Are you willing to help with this request?

Maybe (please elaborate above)

**Comment by bkchr** [comment]
Do you have a little bit more context?

**Comment by StackOverflowExcept1on** [comment]
@bkchr Rust 1.82 (stable Oct 17) enables `multi-value` and `reference-types` by default. I think it makes sense to additionally disable these features.

**Comment by nazar-pc** [comment]
According to link you have provided `-C target-cpu=mvp` is already sufficient to disable all of these features, not sure why Substrate also has `-C target-feature=-sign-ext` given that same document explicitly mentions that it'll be disabled for `mvp` CPU.

**Comment by StackOverflowExcept1on** [comment]
Yep, maybe it make sense just leave `-C target-cpu=mvp ` since Rust added more documentation about this stuff

**Comment by bkchr** [comment]
> not sure why Substrate also has `-C target-feature=-sign-ext` given that same document explicitly mentions that it'll be disabled for `mvp` CPU.

Apparently it did not worked before: https://github.com/paritytech/substrate/pull/13804#issue-1651514636

AFAIR we also just disabled this feature because there were issues with our wasm parser and not because we didn't wanted this feature.

**Comment by StackOverflowExcept1on** [comment]
@bkchr @koute We could switch to the new target [`wasm32v1-none`](https://doc.rust-lang.org/nightly/rustc/platform-support/wasm32v1-none.html) (it will be stabilized starting with Rust 1.84 and no more hacks with `RUSTC_BOOTSTRAP=1`) instead of `wasm32-unknown-unknown`. I've now updated to `nightly-2024-12-14` and we can't build gear smart contracts even with `RUSTFLAGS=-Ctarget-cpu=mvp` and `-Zbuild-std`. Although runtime is building on the latest nightly version so far.

**Comment by bkchr** [comment]
Sounds reasonable

**Comment by StackOverflowExcept1on** [comment]
@bkchr @koute I could do PR that would change the target to `wasm32v1-none`, but it doesn't have `std` as `wasm32-unknown-unknown`. Also, I don't fully understand why every runtime has this line: https://github.com/paritytech/polkadot-sdk/blob/bd2c35fcf18614c9012117edc21acce0f653a5fa/templates/minimal/runtime/src/lib.rs#L20
I.e. does this mean that the runtime requires std? In which cases does it use std and in which cases does it not?

**Comment by bkchr** [comment]
> I.e. does this mean that the runtime requires std? In which cases does it use std and in which cases does it not?

No, the piece of code you have linked means that if `std` feature is not set, `no_std` is set to announce to rustc that `std` is not required.



> I could do PR that would change the target to `wasm32v1-none`, but it doesn't have `std` as `wasm32-unknown-unknown`.

This would need to be done in a backwards compatible way, aka only do so if the new target exists.
